docs/products/clickhouse/howto/create-dictionary.md

```diff
@@ -0,0 +1,260 @@
+  the permissions of the `avnadmin` user even if another user creates the dictionary.
+  In upstream ClickHouse, the same is true except the `default` user is used.
+
+### Supported layouts
+
+Aiven for ClickHouse supports the same
+[layouts that the upstream ClickHouse supports](https://clickhouse.com/docs/en/sql-reference/dictionaries#ways-to-store-dictionaries-in-memory)
+with two exceptions,`ssd_cache` and `complex_key_ssd_cache`, which are not supported.
+
+### Supported sources
+
+- HTTP(s)
+- Upstream/remote ClickHouse
+- Aiven for ClickHouse
+
+<!--
+- Upstream/remote MySQL®
+- Aiven for MySQL
+- Upstream/remote PostgreSQL®
+- Aiven for PostgreSQL
+-->
+
+## Create a dictionary
+
+To create a dictionary with specified structure (attributes), source, layout, and lifetime,
+use the following syntax:
+
+```sql
+CREATE [OR REPLACE] DICTIONARY [IF NOT EXISTS] [db.]dictionary_name
+(
+    key1 type1  [DEFAULT|EXPRESSION expr1] [IS_OBJECT_ID],
+    key2 type2  [DEFAULT|EXPRESSION expr2],
+    attr1 type2 [DEFAULT|EXPRESSION expr3] [HIERARCHICAL|INJECTIVE],
+    attr2 type2 [DEFAULT|EXPRESSION expr4] [HIERARCHICAL|INJECTIVE]
+)
+PRIMARY KEY key1, key2
+SOURCE(SOURCE_NAME([param1 value1 ... paramN valueN]))
+LAYOUT(LAYOUT_NAME([param_name param_value]))
+LIFETIME({MIN min_val MAX max_val | max_val})
+SETTINGS(setting_name = setting_value, setting_name = setting_value, ...)
+COMMENT 'Comment'
+```
+
+## Examples
+
+### Speeding up `JOIN`s
+
+1. Create tables in your ClickHouse database:
+
+    ```sql
+    CREATE TABLE users
+    (
+        id UInt64,
+        username String,
+        email String,
+        country String
+    )
+    ENGINE = MergeTree()
+    ORDER BY id;
+    ```
+
+    ```sql
+    CREATE TABLE transactions
+    (
+        id UInt64,
+        user_id UInt64,
+        product_id UInt64,
+        quantity Float64,
+        price Float64
+    )
+    ENGINE = MergeTree()
+    ORDER BY id;
+    ```
+
+1. Create a dictionary for the `users` table:
+
+    ```sql
+    CREATE DICTIONARY users_dictionary
+    (
+        id UInt64,
+        username String,
+        email String,
+        country String
+    )
+    PRIMARY KEY id
+    SOURCE(CLICKHOUSE(DB 'default' TABLE 'users'))
+    LAYOUT(FLAT())
+    LIFETIME(100);
+    ```
+
+    You can do the same using the `QUERY` parameter:
+
+    ```sql
+    CREATE DICTIONARY users_dictionary
+    (
+        id UInt64,
+        username String,
+        email String,
+        country String
+    )
+    PRIMARY KEY id
+    SOURCE(CLICKHOUSE(QUERY 'SELECT id, username, email, country FROM default.users'))
+    LAYOUT(FLAT())
+    LIFETIME(100);
+    ```
+
+`JOIN`s are much faster as the data is pre-indexed in memory.
+
+```sql
+SELECT
+    t.id,
+    u.username,
+    t.product_id,
+    t.quantity,
+    t.price
+FROM transactions AS t
+ANY LEFT JOIN users_dictionary AS u
+ON t.user_id = u.id;
+```
+
+### Caching data from an external database / URL
```

Comment:
To align with the style guide: https://developers.google.com/style/slashes#slashes-with-alternatives
```suggestion
### Caching data from an external database or URL
```